x-pack/plugins/security_solution/server/lib/detection_engine/rule_response_actions/schedule_notification_response_actions.ts

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { map, uniq } from 'lodash';
+import type { RuleResponseAction } from '../../../../common/detection_engine/rule_response_actions/schemas';
+import { RESPONSE_ACTION_TYPES } from '../../../../common/detection_engine/rule_response_actions/schemas';
+import type { SetupPlugins } from '../../../plugin_contract';
+
+interface OsqueryQuery {
+  id: string;
+  query: string;
+  ecs_mapping: Record<string, Record<'field', string>>;
+  version: string;
+  platform: string;
+}
+
+interface OsqueryResponseAction {
+  actionTypeId: RESPONSE_ACTION_TYPES.OSQUERY;
+  params: {
+    id: string;
+    queries: OsqueryQuery[];
+    savedQueryId: string;
+    query: string;
+    packId: string;
+    ecs_mapping?: Record<string, { field?: string; value?: string }>;
+  };
+}
+
+interface ScheduleNotificationActions {
+  signals: unknown[];
```

Comment:
Nit: any reason why not call this `alerts` instead of the deprecated `signals`?

Reply:
Hmm, good question. For the purpose of this component, I think we could call it `alerts`, but there are some other places where these are signals, so I didn't want to stand out ;p 
eg. `x-pack/plugins/security_solution/server/lib/detection_engine/rule_types/create_security_rule_type_wrapper.ts`